internal/oracle/utils.go

```diff
@@ -65,12 +65,18 @@ func (validator *YieldedValidator) calculateRating(currentBlock uint64) (int64,
 	if err != nil {
 		log.Fatalf("Failed to parse yield: %v", err)
 	}
-	commission_percentage := commission_float / float64(100000)
+	commission_percentage := commission_float / float64(10000)
 	adjusted_apy := (1 - commission_percentage) * yield_float * 100
 	continuity := float64(validator.PbftCount) / float64(currentBlock-validator.RegistrationBlock)
 
-	//w1 * (APY) - (Commission * w2) + w3 * Continuity + w4 * stake
-	score := float64(0.4)*adjusted_apy - float64(0.1)*commission_float + float64(0.5)*continuity
+	// Check for 100% commission and zero yield
+	score := float64(0)
+	if commission_float == 10000 || yield_float == 0 {
+		score = 0
+	} else {
+		//w1 * (APY) - (Commission * w2) + w3 * Continuity + w4 * stake
+		score = float64(0.4)*adjusted_apy - float64(0.1)*commission_float + float64(0.5)*continuity
+	}
```

Comment:
Let's just make it simpler for now and use `(1 - commission_percentage) * yield_float * 100`. We can add more conditions later.

Reply:
Also, for the testnet, it makes sense to not send nodes that have 100% delegation to the contract (as that's all the current ones) and so the tx won't be so big - https://testnet.explorer.taraxa.io/tx/0x8cda0ba0c6db9e189634c14455173aa3adb4d9fa236377e25407ead3bb6b6066